app/amity.py

```diff
@@ -0,0 +1,466 @@
+            print("\nFellows")
+            print(name + " - Fellow")
+        print("\n")
+
+    def print_unallocated(self, file_name=None):
+        office_waiting = len(self.office_waiting_list)
+        living_space_waiting = len(self.living_space_waiting_list)
+
+        empl_msg = "\nEmployees waiting for office: %s \n" % office_waiting
+        print(empl_msg)
+        # return empl_msg
+        for employee in self.office_waiting_list:
+            print(employee)
+            # return employee
+
+        fell_msg = "\nFellow waiting living space: %s\n" % living_space_waiting
+        print(fell_msg)
+        # return fell_msg
+        for fellow in self.living_space_waiting_list:
+            print(fellow)
+            # return fellow
+
+        if file_name is not None:
+            # temp = file_name.split(".")
+            # if len(temp) == 1:
+            print("\n Printing out to file - ", file_name)
+            # return("\n Printing out to file - ", file_name)
+            with open(file_name, 'a') as f:
+
+                # write out all employees without office
+                time_stamp = time.strftime("%Y-%m-%d %H:%M")
+                f.write("\nUnallocated as of " + time_stamp)
+                f.write(empl_msg)
+                for employee in self.office_waiting_list:
+                    f.write((employee) + "\n")
+
+                # write out fellows without living space
+                f.write(fell_msg)
+                for fellow in self.living_space_waiting_list:
+                    f.write((fellow) + "\n")
+            # else:
+                # print("Please indicate a valid file name and extension")
+                # return("Please indicate a valid file name and extension")
+
+    def print_data(self, room_name):
+        room_name = room_name.upper()
+        global room_type
+
+        if room_name not in self.all_rooms:
+            print("Sorry! Room is none existent")
+        else:
+            # check if room exists and is an office
+            if room_name in self.all_offices:
+                room_type = "office"
+            elif room_name in self.all_living:
+                room_type = "living"
+
+            # print out room name and type
+            room_details = (room_name + " - " + room_type)
+            h_line = ('-' * 30)
+            # print out occupants
+            occupants = (',    '.join(self.room_allocations[room_type][room_name]))
+            if len(self.room_allocations[room_type][room_name]) == 0:
+                occupants = "No occupants"
+            msg = (h_line + "\n" + room_details + "\n" + h_line + "\n" + occupants + "\n\n")
+        # print(msg)
+        return(msg)
+
+    def print_room(self, room_name):
+        print(self.print_data(room_name))
+
+    def print_allocations(self, file_name=""):
+        for room in self.all_rooms:
+            allocations = self.print_data(room)
+            print(allocations)
+
+        if file_name is not None or "":
```

Comment:
This statement could be shorter and easily readable i.e. `if file_name`. Refer to *truthy* and *falsy* values for context.